chore: add engines to package.json

ju-Skinner opened the pull request:
# Description
This assists developers with knowing which node versions are compatible with the Design System. 


Please delete options that are not relevant. 
If your type of change is not present, add that option.
- [X] Chore (other changes that don't affect source or test files)

# How Has This Been Tested?
N/A

**Test Configuration**:

- Sage versions:
- OS:
- Browsers:
- Screen readers:
- Misc:

# Checklist:

If not applicable, leave options unchecked.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR
- [X] No options apply

monkeypox8 commented on app/assets/stylesheets/sage/patterns/elements/_checkbox.scss:
```diff
@@ -0,0 +1,121 @@
+    color: $sage-checkbox-color-default;
+    cursor: pointer;
+    display: inline-block;
+    height: 1.5rem;
```
Totally optional, but you might consider making this (and the width) a variable so both can be assigned at once. It could make theming a little easier down the road.

monkeypox8 commented on app/assets/stylesheets/sage/patterns/elements/_checkbox.scss:
```diff
@@ -0,0 +1,121 @@
+      box-shadow: sage-shadow(sm);
+      &::after {
+        content: "";
+        border: 2px solid #fff;
```
🔥Love this. We'll probably want to convert most of these values to variables, but it can wait for a follow-up PR when adding the Nucelo animation.

ju-Skinner committed:
chore: add engines to package.json

this assists developers with knowing which node versions are compatible

voodooGQ commented on package.json:
```diff
@@ -1,10 +1,12 @@
     "libs/*"
   ],
+  "engines": {
+    "node": ">= 16"
```
Any reason we're starting with 16 instead of 18? 16 stopped receiving security support on 10-18-2022

ju-Skinner replied:
Yes, you are right. Does that mean that people should be forced to use 18 to develop within our ecosystem? If I recall, we are saying that we know that our code works with this minimum version of node. End of Life for 16 is later this year, you can find out more [here](https://nodejs.dev/en/about/releases/)

voodooGQ replied:
You're correct, but with that stance we have to have some sort of testing scenarios in place to verify 16 and 18 both work (probably from CI). If that's in the plans then I think this is totally reasonable, if not I'd say that we should probably put it on the docket.

ju-Skinner replied:
See the latest [commit](https://github.com/Kajabi/sage/pull/50/commits/a94f8d2ec4e3ddead98d190bdf831b054eabc509)

ju-Skinner committed:
ci: add matrix strategy for node versions we support

ju-Skinner committed:
ci: update matrix to include node version 20

You can find the values of the variable under the Repo Settings ->
Secrets and Variables -> Actions
